api/client/connection.go

```diff
@@ -289,6 +290,40 @@ func (c *Connection) Load(ctx context.Context, poolID ksuid.KSUID, branchName st
 	return commit, err
 }
 
+func (c *Connection) IndexRulesPost(ctx context.Context, rules []index.Rule) error {
+	req := c.NewRequest(ctx, http.MethodPost, "/index", rules)
+	res, err := c.Do(req)
+	res.Body.Close()
+	return err
+}
+
+func (c *Connection) IndexRulesDelete(ctx context.Context, ids []ksuid.KSUID) (api.IndexRulesDeleteResponse, error) {
+	var request api.IndexRulesDeleteRequest
+	for _, id := range ids {
+		request.RuleIDs = append(request.RuleIDs, id.String())
+	}
+	req := c.NewRequest(ctx, http.MethodDelete, "/index", request)
+	var deleted api.IndexRulesDeleteResponse
+	err := c.doAndUnmarshal(req, &deleted)
+	return deleted, err
+}
+
+func (c *Connection) IndexApply(ctx context.Context, poolID ksuid.KSUID, branchName, rule string, oids []ksuid.KSUID) (api.CommitResponse, error) {
+	path := urlPath("/pool", poolID.String(), "branch", branchName, "index")
+	req := c.NewRequest(ctx, http.MethodPost, path, api.ApplyIndexRequest{RuleName: rule, Tags: oids})
+	var commit api.CommitResponse
+	err := c.doAndUnmarshal(req, &commit)
+	return commit, err
+}
+
+func (c *Connection) IndexUpdate(ctx context.Context, poolID ksuid.KSUID, branchName string, rules []string) (api.CommitResponse, error) {
+	path := urlPath("/pool", poolID.String(), "branch", branchName, "index/update")
+	req := c.NewRequest(ctx, http.MethodPost, path, api.UpdateIndexRequest{RuleNames: rules})
+	var commit api.CommitResponse
+	err := c.doAndUnmarshal(req, &commit)
+	return commit, err
+}
+
```

Comment:
Nit: Put these after Connection.Delete so that method stays adjacent to the non-index branch methods.